registry/remote/repository.go

```diff
@@ -544,10 +542,21 @@ func (r *Repository) referrersPageByAPI(ctx context.Context, artifactType string
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
+	switch resp.StatusCode {
+	case http.StatusOK:
+	case http.StatusNotFound:
+		// A 404 returned by Referrers API indicates that Referrers API is not supported.
+		return "", errors.Join(errdef.ErrUnsupported, errutil.ParseErrorResponse(resp))
```

Comment:
If the status code is 404 and the error code is `NAME_UNKNOWN`, it means that the repository does not exist, and the referrers API status is still unknown. For such case, we shouldn't return `ErrUnsupported`.

Reply:
How about:

```suggestion
		if errResp := errutil.ParseErrorResponse(resp); errutil.IsErrorCode(errResp, errcode.ErrorCodeNameUnknown) {
			// The repository is not found, Referrers API status is unknown
			return "", errResp
		}
		// Referrers API is not supported.
		return "", fmt.Errorf("failed to query referrers API: %w", errdef.ErrUnsupported)
```